src/lib.rs

```diff
@@ -265,6 +265,17 @@ impl DXGIManager {
         }
     }
 
+    pub fn geometry(&mut self) -> (usize, usize) {
+        let output_desc = self.duplicated_output.as_mut().unwrap().get_desc();
```

Comment:
would just require changing this `.as_mut()` to `.as_ref()`

Reply:
See, I didn't know that! :O I'll change that and see if it works so we can merge this bad boy